test/irmin-pack/test_repo.ml

```diff
@@ -0,0 +1,54 @@
+(*
+ * Copyright (c) 2022-2022 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open! Import
+open Common
+
+let src = Logs.Src.create "tests.repo" ~doc:"Test Repo"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+let test_dir = "_build"
+
+let fresh_name =
+  let c = ref 0 in
+  fun () ->
+    incr c;
+    let name = Filename.concat test_dir ("test-repo" ^ string_of_int !c) in
+    name
+
+include struct
+  module Maker = Irmin_pack_unix.Maker (Conf)
+  module Store = Maker.Make (Schema)
+
+  let config ?(readonly = false) ?root () =
+    let root = match root with None -> fresh_name () | Some x -> x in
+    Irmin_pack.config ~readonly
+      ~indexing_strategy:Irmin_pack.Indexing_strategy.minimal root
+end
+
+let test_readonly_batch () =
+  let config = config ~readonly:true () in
+  let* repo = Store.Repo.v config in
```

Comment:
I think the CI is failing because you cannot open an empty store in RO. 

There is a file `multiple_instances.ml` which is used to test the RO instances, so you could reuse it, and maybe rename it in something better :)

Reply:
Ah, I tricked myself locally by having a read-write store first. I'll think about how to rework this.